### PR TITLE
update unit tests to use localized resources

### LINKED ISSUE #821: Roslyn.Services.Editor.CSharp.UnitTests.dll - 307 tests fail with Russian language set as primary with localized resources present
Ported from TFS WorkItem: **1101388**

------------------------------

**Repro Steps:**

**Steps to reproduce (likely to&nbsp;repro with other languages):**<BR>

1. Follow three steps described at [http://windows.microsoft.com/en-us/windows-8/using-multiple-languages](http://windows.microsoft.com/en-us/windows-8/using-multiple-languages)&nbsp;to install Russian language pack and make Russian the primary language.

&nbsp;

2. Queue Main-PerCheckin-Debug build with pseudo localization for Russian language (go to the Parameters tab in the Queue Build dialog and change Localization Languages value to RUS)

&nbsp;

3. Wait for the drop. Copy binaries to the machine prepared on step 1. Copy localize\RUS folder from the drop location next to the binaries and rename it to 'ru'. Verify that output of command line compilers is localized.

&nbsp;

4. Run tests. I simply copy bits over to Binaries\Debug folder on the previous step and run 'test' script.

&nbsp;

The test log is attached.

------------------------------

**Revisions:**

1) Created By Aleksey Tsingauz (1/2/2015 10:22:53 AM)

------------------------------

### PR DESCRIPTION
**Pull request by brettfo**
Also fixes some stray whitespace in a C# resource (`CSharpFeaturesResources.DisposePattern`) and moves duplicated resources from `CSharpFeaturesResources` and `VBFeaturesResources` to `WorkspacesResources.Usage` and updates the product code to actually use this resource in `ISymbolExtensions.cs` (it was hard-coded English before.)

Fixes #821.

### CODE REVIEW

**Comment by brettfo** [comment]
Potential reviewers: @Pilchie @jasonmalinowski @dpoeschl @rchande @balajikris @basoundr

File: src/Features/CSharp/CSharpFeaturesResources.resx [hunk 1]
```diff
@@ -279,7 +276,7 @@ private bool disposedValue = false; // To detect redundant calls
     }}
 }}
 
- // TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources. 
+// TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
```
Comment (dpoeschl) [inline src/Features/CSharp/CSharpFeaturesResources.resx]: I assume this mistake just got formatted away when actually inserted?

**Comment by dpoeschl** [comment]
LGTM. Tagging @shyamnamboodiripad who I think did the IDisposable feature (or will know who did), which has the most interesting changes.

**Comment by brettfo** [comment]
@dpoeschl Regarding the whitespace; yes it got formatted away.  And as for the changes around IDisposable, for C# I just fetch the resource string then manually indent it to match the expected output, and for VB I fetch the resource then do some manual simplification (e.g., remove unnecessary `Me.` before a property), but the functionality of the tests hasn't changed.

**Comment by jaredpar** [comment]
@dotnet-bot test this please

**Pull request merged**